portfolio/src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -40,27 +43,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     JSONArray eventsArray = jsonFromRequest.getJSONArray("events");
     JSONArray tasksArray = jsonFromRequest.getJSONArray("tasks");
-    String workHoursStartTimeString = (String) jsonFromRequest.get("startTime");
-    String workHoursEndTimeString = (String) jsonFromRequest.get("endTime");
+    String workHoursStartTimeString = jsonFromRequest.getString("startTime");
+    String workHoursEndTimeString = jsonFromRequest.getString("endTime");
     Instant workHoursStartTime = Instant.parse(workHoursStartTimeString);
     Instant workHoursEndTime = Instant.parse(workHoursEndTimeString);
-
+    String algorithmTypeString = jsonFromRequest.getString("algorithmType");
     Collection<CalendarEvent> events = collectEventsFromJsonArray(eventsArray);
     Collection<Task> tasks = collectTasksFromJsonArray(tasksArray);
+
+    Optional<SchedulingAlgorithmType> schedulingAlgorithmTypeOptional =
+        getSchedulingAlgorithmTypeOptional(algorithmTypeString);
+    if (!schedulingAlgorithmTypeOptional.isPresent()) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "The request by the client was syntactically incorrect.");
```

Comment:
Consider clarifying that the algorithm type could not be determined.

Reply:
Done.